Library/Homebrew/test/cask/cli/upgrade_spec.rb

```diff
@@ -0,0 +1,219 @@
+      local_transmission_route = Hbc.appdir.join("Transmission.app")
+
+      expect(local_caffeine).to be_installed
+      expect(local_caffeine_route).to be_a_directory
+      expect(local_caffeine.versions).to include("1.2.2")
+
+      expect(local_transmission).to be_installed
+      expect(local_transmission_route).to be_a_directory
+      expect(local_transmission.versions).to include("2.60")
+
+      described_class.run
+
+      expect(local_caffeine).to be_installed
+      expect(local_caffeine_route).to be_a_directory
+      expect(local_caffeine.versions).to include("1.2.3")
+
+      expect(local_transmission).to be_installed
+      expect(local_transmission_route).to be_a_directory
+      expect(local_transmission.versions).to include("2.61")
+    end
+
+    it "and updates only the Casks specified in the command line" do
+      local_caffeine = Hbc::CaskLoader.load("local-caffeine")
+      local_caffeine_route = Hbc.appdir.join("Caffeine.app")
+      local_transmission = Hbc::CaskLoader.load("local-transmission")
+      local_transmission_route = Hbc.appdir.join("Transmission.app")
+
+      expect(local_caffeine).to be_installed
+      expect(local_caffeine_route).to be_a_directory
+      expect(local_caffeine.versions).to include("1.2.2")
+
+      expect(local_transmission).to be_installed
+      expect(local_transmission_route).to be_a_directory
+      expect(local_transmission.versions).to include("2.60")
+
+      described_class.run("local-caffeine")
+
+      expect(local_caffeine).to be_installed
+      expect(local_caffeine_route).to be_a_directory
+      expect(local_caffeine.versions).to include("1.2.3")
+
+      expect(local_transmission).to be_installed
+      expect(local_transmission_route).to be_a_directory
+      expect(local_transmission.versions).to include("2.60")
+    end
+
+    it 'updates "auto_updates" and "latest" Casks when their tokens are provided in the command line' do
+      local_caffeine = Hbc::CaskLoader.load("local-caffeine")
+      local_caffeine_route = Hbc.appdir.join("Caffeine.app")
+      auto_updates = Hbc::CaskLoader.load("auto-updates")
+      auto_updates_path = Hbc.appdir.join("MyFancyApp.app")
+
+      expect(local_caffeine).to be_installed
+      expect(local_caffeine_route).to be_a_directory
+      expect(local_caffeine.versions).to include("1.2.2")
+
+      expect(auto_updates).to be_installed
+      expect(auto_updates_path).to be_a_directory
+      expect(auto_updates.versions).to include("2.57")
+
+      described_class.run("local-caffeine", "auto-updates")
+
+      expect(local_caffeine).to be_installed
+      expect(local_caffeine_route).to be_a_directory
+      expect(local_caffeine.versions).to include("1.2.3")
+
+      expect(auto_updates).to be_installed
+      expect(auto_updates_path).to be_a_directory
+      expect(auto_updates.versions).to include("2.61")
+    end
+  end
+
+  describe "with --greedy it checks additional Casks" do
+    include_context "Proper Casks"
```

Comment:
Instead of including this twice, create a super-context, i.e. `context "successful upgrade" do`.